src/mm-input-mask/mm-input-mask.js

```diff
@@ -0,0 +1,396 @@
+/**
+ * @license
+ * Copyright (c) 2015 MediaMath Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at http://mediamath.github.io/strand/LICENSE.txt
+
+*/
+(function () {
+
+	_types = {
+		SEP:0,
+		GROUP:1
+	};
+
+	_restrict = {
+		alpha:/[A-Za-z]+/g,
+		numeric:/\d+/ig,
+		alphaNumeric:/[A-Za-z\d]+/g,
+		all:/.+/g
+	};
+
+	_applyReg = function(reg, val) {
+		if (reg) {
+			reg.lastIndex = 0;
+			return reg.test(val);
+		}
+		return false;
+	};
+
+	_rules = {
+		hour12:function(char, input, group, field) {
+
+			var val = input+char,
+				first = val.length === 1,
+				second = val.length === 2,
+				imp = parseInt(char);
+
+			if (first) {
+				if (imp > 1) {
+					field.value = "0" + input;
+					return false;
+				}
+			} else if (second) {
+				if (imp > 2 && parseInt(input) === 1) {
+					field.value = input + "2";
+					return false;
+				}
+			}
+			return true;
+		},
+		hour24: function(char, input, group, field) {
+			var val = input+char,
+				first = val.length === 1,
+				second = val.length === 2,
+				imp = parseInt(char);
+
+		},
+		minutes: function(char, input, group, field) {
```

Comment:
Some of these rules seem kind of weird...  For instance, in the minutes input, if you delete the current value an enter the following keys:
- "7"
- tab
you will get "05" minutes, but if you enter for instance:
- "7"
- left arrow
- "8"
- tab
you will get "85" minutes

Further, the rules get applied but overwritten when you have text in the inputs selected/highlighted.

(Also, forward tabbing seems to be in an infinite loop, and reverse tabbing seems to dead-end)